metricbeat/module/nifi/flowfilerepostorage/data.go

```diff
@@ -0,0 +1,55 @@
+package flowfilerepostorage
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+// FlowFileRepositoryStorageUsage ...
+type FlowFileRepositoryStorageUsage struct {
+	FreeSpace       string `json:"freeSpace"`
+	FreeSpaceBytes  uint64 `json:"freeSpaceBytes"`
+	TotalSpace      string `json:"totalSpace"`
+	TotalSpaceBytes uint64 `json:"totalSpaceBytes"`
+	UsedSpace       string `json:"usedSpace"`
+	UsedSpaceBytes  uint64 `json:"usedSpaceBytes"`
+	Utilization     string `json:"utilization"`
+}
+
+// Data ...
+type Data struct {
+	SystemDiagnostics struct {
+		AggregateSnapshot struct {
+			FlowFileRepositoryStorageUsage `json:"flowFileRepositoryStorageUsage"`
+		} `json:"aggregateSnapshot"`
+	} `json:"systemDiagnostics"`
+}
+
+func eventMapping(body io.Reader) common.MapStr {
+	var data Data
+	err := json.NewDecoder(body).Decode(&data)
+	if err != nil {
+		logp.Err("Error: ", err)
+	}
+
+	slice := data.SystemDiagnostics.AggregateSnapshot.FlowFileRepositoryStorageUsage
+
+	fmt.Printf("%v", data)
+	fmt.Printf("%v", slice)
+
+	event := common.MapStr{
```

Comment:
Looking at couchbase, I would today to a combination of schema and using a struct. We should have a schema for the parts under the arrays and use the struct to handle the arrays, because there is no support for arrays at the moment (@tsg correct me if I'm wrong). Here is an example for more heavy json restructuring with schema:

```
var (
	schema = s.Schema{
		"system_diagnostics": c.Dict("systemDiagnostics", s.Schema{
			"aggregate_snapshot": c.Dict("aggregateSnapshot", s.Schema{
				"max_heap": s.Object{
					"bytes": c.Int("maxHeapBytes"),
				},
				"available_processors": c.Int("availableProcessors"),
			}),
		}),
	}
)
```

I mainly wanted to show haw dict and object can be combined to get the correct trees. It does not show an example for the arrays. 

Looking at all the arrays like `garbageCollection` or `contentRepositoryStorageUsage` it will get interesting on how to handle this. One generic recommendation I have here is always to think on how people will want to query this information.